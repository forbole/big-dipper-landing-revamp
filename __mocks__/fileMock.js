// __mocks__/fileMock.js
module.exports = {
  blurDataURL: 'data:image/png;base64,imagedata',
  height: 24,
  src: '/img.jpg',
  width: 24,
}