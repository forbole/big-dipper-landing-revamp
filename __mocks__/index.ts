export { default as MockTheme } from "./theme";
