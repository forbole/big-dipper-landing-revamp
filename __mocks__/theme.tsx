/* eslint-disable */
import { ThemeProvider } from '@mui/material/styles';
import { ReactNode } from 'react';
import { darkTheme } from '@/src/styles/theme';

/**
 * Theme mocker to handle custom keys
 */
const MockTheme = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={darkTheme}>{children}</ThemeProvider>;
};

export default MockTheme;
