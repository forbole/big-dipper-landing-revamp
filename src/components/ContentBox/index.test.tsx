import { MockTheme } from "@/__mocks__";
import { render, screen } from "@testing-library/react";

import Component from ".";

// ==================================
// unit tests
// ==================================
describe("ContentBox", () => {
  it("matches snapshot", async () => {
    const { container } = render(
      <MockTheme>
        <Component>
          <div>HELLO WORLD</div>
        </Component>
      </MockTheme>,
    );

    expect(screen.getByText(/HELLO WORLD/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
