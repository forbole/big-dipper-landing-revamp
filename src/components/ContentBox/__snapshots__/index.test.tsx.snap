// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBox matches snapshot 1`] = `
.emotion-0 {
  background-color: rgba(255, 255, 255, 0.5);
  border: solid #fff 2px;
  border-radius: 0.75rem;
  color: #000;
  overflow: auto;
  padding: 32px;
}

@media (min-width:768px) {
  .emotion-0 {
    padding: 32px;
  }
}

<div>
  <div
    class="contentbox__container MuiBox-root emotion-0"
  >
    <div>
      HELLO WORLD
    </div>
  </div>
</div>
`;
